docs/walletkit/android/chain-abstraction.mdx

```diff
@@ -0,0 +1,233 @@
+                }
+            )
+          }
+
+          is Wallet.Model.FulfilmentSuccess.NotRequired -> {
+              //Chain abstraction is not required, handle transaction as usual
+          }
+        }
+      },
+      onError = { fulfilmentError ->
+            // One of the possible errors: NoRoutesAvailable, InsufficientFunds, InsufficientGasFunds - wallet should send the JsonRpcError to a dapp for given request and display error to the user
+      }
+   )
+  } else {
+    // Handle transaction as usual
+  }
+}
+```
+## Types
+
+Following are the types that are used in the chain abstraction methods.
+
+```kotlin
+data class Transaction(
+  var from: String,
+  var to: String,
+  var value: String,
+  var gas: String,
+  var gasPrice: String,
+  var data: String,
+  var nonce: String,
+  var maxFeePerGas: String,
+  var maxPriorityFeePerGas: String,
+  var chainId: String
+)
+
+sealed class FulfilmentSuccess : Model() {
+  data class Available(
+    val fulfilmentId: String,
+    val checkIn: Long,
+    val transactions: List<Transaction>,
+    val initialTransaction: Transaction,
+    val initialTransactionMetadata: InitialTransactionMetadata,
+    val funding: List<FundingMetadata>
+  ) : FulfilmentSuccess()
+
+  data class NotRequired(val initialTransaction: Transaction) : FulfilmentSuccess()
+}
+
+data class InitialTransactionMetadata(
+  var symbol: String,
+  var amount: String,
+  var decimals: Int,
+  var tokenContract: String,
+  var transferTo: String
+)
+
+data class FundingMetadata(
+  var chainId: String,
+  var tokenContract: String,
+  var symbol: String,
+  var amount: String,
+  var bridgingFee: String,
+  var decimals: Int
+)
+
+sealed class FulfilmentError : Model() {
+  data object NoRoutesAvailable : FulfilmentError()
+  data object InsufficientFunds : FulfilmentError()
+  data object InsufficientGasFunds : FulfilmentError()
+  data class Unknown(val message: String) : FulfilmentError()
+}
+
+sealed class FulfilmentStatus : Model() {
+  data class Completed(val createdAt: Long) : FulfilmentStatus()
+  data class Error(val reason: String) : FulfilmentStatus()
+}
+
+sealed class FulfilmentStatus : Model() {
+  data class Completed(val createdAt: Long) : FulfilmentStatus()
+  data class Error(val reason: String) : FulfilmentStatus()
+}
+
+data class FulfilmentDetails(
+  var fulfilmentDetails: List<TransactionDetails>,
+  var initialDetails: TransactionDetails,
+  var bridgeDetails: List<TransactionFee>,
+  var localTotal: Amount
+)
+
+data class TransactionDetails(
+  var transaction: Transaction,
+  var eip1559: EstimatedFees,
+  var transactionFee: TransactionFee
+)
+
+data class TransactionFee(
+  var fee: Amount,
+  var localFee: Amount
+)
+
+data class Amount(
+  var symbol: String,
+  var amount: String,
+  var unit: String,
+  var formatted: String,
+  var formattedAlt: String
+)
+```
+
+{/* TODO: update sample wallet link */}
```

Comment:
I'll provide this 👌

Reply:
@jakubuid can you provide this, so I can update

Reply:
Here you go: https://appdistribution.firebase.dev/i/076a3bc9669d3bee

Reply:
do we need to link the deployments or the source? imo source makes more sense as folks will be able to clone/build the wallet themselves

Reply:
Yes we should provide the link to the sources as well to showcase the implementation. @vrajdesai78 do we have that already?

> do we need to link the deployments
@ganchoradkov yes because they could get samples and test

Reply:
> Yes we should provide the link to the sources as well to showcase the implementation. @vrajdesai78 do we have that already?
> 
> > do we need to link the deployments
> 
> @ganchoradkov yes because they could get samples and test

I added it in testing section.